---
title: The Epigenetic Machinery
output:
html_document:
  theme: journal
---

<style type="text/css">
.main-container {
  margin-left: 10px;
  margin-right: auto;
}
</style>

```{r, echo=FALSE,results="hide"}
library(DT)
library(readr)
```

```{r prepDF,echo=FALSE,results="hide", message=FALSE}
df <- read_csv("data/EM.csv")
df <- as.data.frame(df)
web <- read_csv("data/EM_genes_website.csv")
web <- as.data.frame(web)

df_basic <- df[, c("Gene_name", "exac.pLI", "coexpression_status")]
names(df_basic) <- c("Gene_Name", "pLI", "Coexprs")
df_basic$pLI <- round(df_basic$pLI, 2)
df_basic$Coexprs <- as.factor(df_basic$Coexprs)
levels(df_basic$Coexprs) <- sub("coexpressed", "coexp", levels(df_basic$Coexprs))

df_func <- df[, c("Epi_function", "Enzymatic.activity", "Reading.activity", "TF_activity",
                  "is.MDEM", "is.neuro.associated", "is.ca.associated")]
names(df_func) <- c("Epig_Function", "Enzy_Activity", "Read_Activity", "TF_Activity", "Is_MDEM", "Is_Neuro_Assoc", "Is_Cancer_Assoc")
df_func$Epig_Function <- as.factor(df_func$Epig_Function)
levels(df_func$Epig_Function) <- gsub(";", "; ", levels(df_func$Epig_Function))
df_func$Enzy_Activity <- as.factor(df_func$Enzy_Activity)
df_func$Read_Activity <- as.factor(df_func$Read_Activity)
levels(df_func$Read_Activity) <- gsub(";", "; ", levels(df_func$Read_Activity))
df_func$TF_Activity <- as.factor(ifelse(df_func$TF_Activity, "TF", "Not TF"))
df_func$Is_MDEM <- as.factor(ifelse(df_func$Is_MDEM, "MDEM", "Not MDEM"))
df_func$Is_Neuro_Assoc <- as.factor(ifelse(df_func$Is_Neuro_Assoc, "Neuro", "Not Neuro"))
df_func$Is_Cancer_Assoc <- as.factor(ifelse(df_func$Is_Cancer_Assoc, "Cancer", "Not Cancer"))

df_names <- df[, c("Alt_Gene_names", "Gene_name_exac", "Gene_name_gtex", "Chr", "Strand", "Start_Pos", "End_Pos")]
names(df_names) <- c("Gene_Name_Alternatives", "Gene_Name_ExAC", "Gene_Name_GTEx", "Chr", "Strand", "Start", "End")

df_omim <- df[, c("OMIM_all_phenotypes"), drop = FALSE]
names(df_omim) <- c("OMIM_All_Phenotypes")

df_protein <- df[, c("Protein.names", "Interpro.Domains", "Entry")]

rownames(web) <- web$Gene_name
web[df_basic$Gene_Name,]
df_web <- data.frame(gnomAD = sprintf('<a href="https://gnomad.broadinstitute.org/gene/%s" target="_blank">%s</a>', web$Ensembl_id, web$Ensembl_id),
                     OMIM = sprintf('<a href="https://www.omim.org/entry/%s" target="_blank">%s</a>', web$mim_number, web$mim_number),
                     GTEx = sprintf('<a href="https://gtexportal.org/home/gene/%s" target="_blank">%s</a>', df_names$Gene_Name_GTEx,  df_names$Gene_Name_GTEx),
                     COSMIC = sprintf('<a href="https://cancer.sanger.ac.uk/cosmic/gene/analysis?ln=%s" target="_blank">%s</a>', web$Gene_name, web$Gene_name),
                     UniProt = sprintf('<a href="https://www.uniprot.org/uniprot/%s" target="_blank">%s</a>', df_protein$Entry, df_protein$Entry),
                     Interpro = sprintf('<a href="https://www.ebi.ac.uk/interpro/protein/%s" target="_blank">%s</a>', df_protein$Entry, df_protein$Entry),
                     
                     stringsAsFactors = FALSE)
df_print <- cbind(df_basic, df_func, df_omim, df_names, df_web)
df_print <- df_print[order(df_print$Gene_Name),]
```

```{r display, echo=FALSE}
datatable(df_print, filter = "top", rownames = FALSE, escape = FALSE,
          extensions = c("Buttons", "ColReorder", "FixedHeader"),
          colnames = c("Gene Name" = 1, "Epigenetic Function" = 4,
                       "Enzymatic Activity" = 5, 
                       "Reader Activity" = 6, "TF Activity" = 7),
          options = list(dom = "Bfrtip",
                         colReorder = TRUE,
                         fixedHeader = TRUE,
                         pageLength = 300,
                         columnDefs = list(list(targets = c(7:10, 11:13, 15:17), visible = FALSE)),
                         buttons = list(c("colvis", "csv"))
                         ))
### buttons = list(list(extend = "columnsToggle",
###                     columns = 1:7))
### buttons = list(c("csv", "colvis"),
###                list(extend = "colvisGroup",
###                            text = "Disease",
###                            hide = 7:10
```

This website provides a curated list of epigenetic machinery genes (that is, genes encoding for epigenetic regulators) as described in

> L Boukas, JM Havrilla, PF Hickey, AR Quinlan, HT Bjornsson, KD Hansen.  
> _Co-expression patterns define epigenetic regulators associated with neurological dysfunction._  
> Genome Res 29: 532â€“542. doi:[10.1101/gr.239442.118](https://doi.org/10.1101/gr.239442.118)

**Column Descriptions:**  
The columns are meant to provide information about basic properties of each of the genes. They are the following:
- **Gene Name** 
- **pLI:** A metric of loss-of-function intolerance (initially derived by the ExAC consortium, now provided by gnomAD), ranging between 0 and 1. The closer it is to 1, the more intolerant to loss-of-function variation a gene is. Note that pLI is essentially a binary measure. 
- **Coexprs:** Co-expression status. In our paper we show that epigenetic machinery genes can be partitioned into 3 categories: 'highly co-expressed', 'co-expressed', and 'not co-expressed', based on their expression patterns in adult human tissues (GTEx). The genes in the highly co-expressed category tend to have very high pLI scores, and are enriched for genes causing neurodevelopmental Mendelian disorders.
- **Epigenetic Function:** Whether the encoded protein is a epigenetic writer, eraser, reader, remodeler, or dual function regulator.
- **Enzymatic activity:** For writers, erasers, and remodelers. For example, whether a writer enzyme is a histone methyltransferase (HMT). 
- **Reader activity:** For readers. For example, whether a reader protein is a histone methylation reader (HMR).
- **TF activity:** Whether the encoded protein has a DNA-binding domain classifying it as a TF as well.
- **Chr:** Chromosome
- **gnomAD:** Link to corresponding gnomAD entry
- **OMIM:** Link to corresponding OMIM entry
- **GTEx:** Link to corresponding GTEx entry
- **COSMIC:** Link to corresponding COSMIC entry
- **UniProt:** Link to corresponding UniProt entry
- **Interpro:** Link to corresponding Interpro entry


